Maximum call stack size exceeded in PFE logs on Codewind

I'm seeing the following in the PFE logs (latest images) when running on Che just after a docker build and push finishes for Generic docker projects:

```
[26/11/19 22:11:31 FileWatcher.js] [ERROR] RangeError: Maximum call stack size exceeded
    at Function.[Symbol.hasInstance] (<anonymous>)
    at Function.isBuffer (buffer.js:430:12)
    at hasBinary (/portal/node_modules/has-binary2/index.js:44:66)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
    at hasBinary (/portal/node_modules/has-binary2/index.js:58:59)
```

It also appears to cause PFE to hang, as it prevented the app and build status from updating. Despite the build finishing and the app deploying, it was stuck in `Building - Creating Image`.

--- a/src/pfe/portal/modules/FileWatcher.js
+++ b/src/pfe/portal/modules/FileWatcher.js
@@ -225,7 +225,7 @@ module.exports = class FileWatcher {
       location: project.projectPath(false),
       applicationPort: project.applicationPort,
       settings: settingsFileContents,
-      language: project.language 
+      language: project.language
     };
 
     log.info(`Calling createProject() for project ${project.name} ${JSON.stringify(projectAction)}`);
@@ -468,7 +468,7 @@ module.exports = class FileWatcher {
       }
       let projectUpdate = { projectID: projectID, projectWatchStateId: projectWatchStateId, ignoredPaths: ignoredPaths };
       await this.handleFWProjectEvent(event, projectUpdate);
-      WebSocket.watchListChanged(data);      
+      WebSocket.watchListChanged(data);
     } catch (err) {
       log.error(err);
     }
@@ -498,7 +498,8 @@ module.exports = class FileWatcher {
         results.error = error;
       }
       let updatedProject = await this.user.projectList.updateProject(projectUpdate);
-      this.user.uiSocket.emit(event, {...results , ...updatedProject});
+      const { logStreams, ...uiProjectInfo } = updatedProject
+      this.user.uiSocket.emit(event, { ...results, ...uiProjectInfo })
       if (fwProject.buildStatus === 'inProgress') {
         // Reset build logs.
         updatedProject.resetLogStream('build');
@@ -666,7 +667,7 @@ function logEvent(event, projectData) {
   }
   log.debug(`${msg} ${projectData.projectID})`);
   log.trace(`${msg}: ${JSON.stringify(projectData, null, 2)})`);
-  
+
   if(updateTimerStart > 0 && event == "projectStatusChanged" && projectData.buildStatus && status == 'success'){
     let updateBuildTimerEnd = Date.now()
     log.info(`${projectData.projectID} update->build end time: ${ updateBuildTimerEnd }`)
@@ -680,5 +681,5 @@ function logEvent(event, projectData) {
     log.info(`${msg} ${projectData.projectID}) total time for update->run: ${ totalUpdateTime } seconds`)
     updateTimerStart = 0;
   }
-  
+
 }

--- a/src/pfe/portal/modules/FileWatcher.js
+++ b/src/pfe/portal/modules/FileWatcher.js
@@ -534,7 +534,8 @@ module.exports = class FileWatcher {
     // We have to emit the full project state *and* the operation status.
     // (Storing the status in the project object is bad as it is
     // only about this close operation.)
-    this.user.uiSocket.emit('projectClosed', {...updatedProject, status: fwProject.status});
+    const { logStreams, ...uiProjectInfo } = updatedProject;
+    this.user.uiSocket.emit('projectClosed', {...uiProjectInfo, status: fwProject.status});
     log.debug('project ' + fwProject.projectID + ' successfully closed');
   }
 

--- a/src/pfe/portal/modules/FileWatcher.js
+++ b/src/pfe/portal/modules/FileWatcher.js
@@ -498,8 +498,9 @@ module.exports = class FileWatcher {
         results.error = error;
       }
       let updatedProject = await this.user.projectList.updateProject(projectUpdate);
-      const { logStreams, ...uiProjectInfo } = updatedProject
-      this.user.uiSocket.emit(event, { ...results, ...uiProjectInfo })
+      // remove fields which are not required by the UI
+      const { logStreams, ...projectInfoForUI } = updatedProject
+      this.user.uiSocket.emit(event, { ...results, ...projectInfoForUI })
       if (fwProject.buildStatus === 'inProgress') {
         // Reset build logs.
         updatedProject.resetLogStream('build');
@@ -534,8 +535,9 @@ module.exports = class FileWatcher {
     // We have to emit the full project state *and* the operation status.
     // (Storing the status in the project object is bad as it is
     // only about this close operation.)
-    const { logStreams, ...uiProjectInfo } = updatedProject;
-    this.user.uiSocket.emit('projectClosed', {...uiProjectInfo, status: fwProject.status});
+    // remove fields which are not required by the UI
+    const { logStreams, ...projectInfoForUI } = updatedProject;
+    this.user.uiSocket.emit('projectClosed', {...projectInfoForUI, status: fwProject.status});
     log.debug('project ' + fwProject.projectID + ' successfully closed');
   }
 

